ctapipe/ml/tools/apply.py

```diff
@@ -0,0 +1,216 @@
+import shutil
+
+import numpy as np
+import tables
+from astropy.table.operations import hstack, vstack
+from tqdm.auto import tqdm
+
+from ctapipe.core.tool import Tool
+from ctapipe.core.traits import Bool, Path, create_class_enum_trait, flag
+from ctapipe.io import TableLoader, write_table
+from ctapipe.io.tableio import TelListToMaskTransform
+
+from ..apply import EnergyRegressor, ParticleIdClassifier
+from ..sklearn import Classifier, Regressor
+from ..stereo_combination import StereoCombiner
+
+
+class Apply(Tool):
+    """Apply machine learning models on data.
+
+    Predict (gamma)-energy and/or particle id.
+    """
+
+    name = "ctapipe-apply"
+    description = __doc__
+
+    overwrite = Bool(default_value=False).tag(config=True)
+
+    input_url = Path(
+        default_value=None,
+        allow_none=False,
+        directory_ok=False,
+        exists=True,
+    ).tag(config=True)
+    output_path = Path(
+        default_value=None,
+        allow_none=False,
+        directory_ok=False,
+    ).tag(config=True)
+
+    energy_regressor_path = Path(
+        default_value=None,
+        allow_none=True,
+        exists=True,
+        directory_ok=False,
+    ).tag(config=True)
+    particle_classifier_path = Path(
+        default_value=None,
+        allow_none=True,
+        exists=True,
+        directory_ok=False,
+    ).tag(config=True)
+
+    stereo_combiner_type = create_class_enum_trait(
+        base_class=StereoCombiner,
+        default_value="StereoMeanCombiner",
+    ).tag(config=True)
+
+    aliases = {
+        ("i", "input"): "Apply.input_url",
+        "regressor": "Apply.energy_regressor_path",
+        "classifier": "Apply.particle_classifier_path",
```

Comment:
With Disp coming to ML, we should change this to energy-regressor and particle-classifier